.github/workflows/build.yml

```diff
@@ -0,0 +1,83 @@
+name: Build all projects with "GCC with Make"
+
+on:
+  push:
+    branches: '*'
+  pull_request:
+    branches: '*'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        project:
+          - cc3220sf
+          - efm32gg
+          - lpc1114fn28
+          - lpc4088
+          - lpc54018
+          - lpc54114
+          - lpc54608
+          - lpc824
+          - mk20d5
+          - mk64f12
+          - mk65f18
+          - mk66f18
+          - mk80f25615
+          - mke15z7
+          - mke18f16
+          - mkl02z4
+          - mkl05z4
+          - mkl25z4
+          - mkl26z4
+          - mkl27z4
+          - mkl27z644
+          - mkl28z7
+          - mkl43z4
+          - mkl46z4
+          - mkv10z7
+          - mkv11z7
+          - mkv31f51212
+          - mkv58f22
+          - mkw01z4
+          - mkw30z4
+          - mkw40z4
+          - mkw41z4
+          - musca_a
+          - musca_b
+          - musca_b_eflash
+          - ncs36510
+          - nrf51xxx
+          - nrf52833
+          - stm32f4xx_2048
+          - stm32l0xx_192
+          - stm32l151
+          - stm32l475
+          - stm32wb55rc
+          - stm32wb55rg
+          - template
+          - tz10xx
+          - w7500
+          # An empty string will build all projects, in case this list is outdated
+          - ""
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    name: Build ${{ matrix.project || 'all' }} on ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+        cache: pip
+    - name: Setup arm-none-eabi-gcc
+      uses: carlosperate/arm-none-eabi-gcc-action@v1
+      with:
+        release: 10.3-2021.10
```

Comment:
Full disclosure, this step uses an Action I maintain, which facilitates Arm GCC installation (with caching, etc), but happy to change it to OS-specific steps if needed.

Reply:
Nice! Glad to hear know about this action!